File: sample_programs/benchmarks/Benchmarks.java
public class Benchmarks {

    public static int sumSquares(int n) {
        int sum = 0;
        for (int i = 1; i <= n; i++) {
            sum = sum + i*i;
        }
        return sum;
    }

    public static void recurse(int depth) {
        if (depth > 0) {
            recurse(depth - 1);
        }
    }

    public static void linkedListInsertTraverse(int numValues) {
        LinkedList list = new LinkedList();
        for (int i = 0; i < numValues; i++) {
            list.append(i);
        }
        for (int i = 0; i < numValues; i++) {
            int result = list.popFirstElement();
        }
    }
}

File: sample_programs/benchmarks/javascript/benchmarks.js
// const LinkedList = require('./linked_list.js');

import { LinkedList } from './linked_list.js';

export function sumSquares(n) {
  let sum = 0;
  for (let i = 1; i <= n; i++) {
    sum += i*i;
  }
  return sum;
}

export function recurse(depth) {
  if (depth > 0) {
    this.recurse(depth - 1);
  }
}

export function linkedListInsertTraverse(numValues) {
  let list = new LinkedList();
  for (let i = 0; i < numValues; i++) {
    list.append(i);
  }
  for (let i = 0; i < numValues; i++) {
    const result = list.popFirstElement();
  }
}

export function traverseArray(size) {
  let array = [];
  for (let i = 0; i < size; i++) {
    array.push(i);
  }
  for (let i = 0; i < size; i++) {
    const temp = array[i];
  }
}
